### ISSUE TITLE
FT 2.0: Improve exception handling when an asynchronous task starts running after an application has shut down

### ISSUE THREAD

**Issue by Azquelt** [OpenLiberty/open-liberty#7208]
This can happen if the asynchronous task took a while to fail and then retries while in the mean time, the app has been shut down.

Related to RTC261239

**Comment by Azquelt** [comment]
Current behaviour is that we get a log and FFDC like this:
```
00000025 com.ibm.ws.microprofile.faulttolerance20.impl.AsyncExecutor  E CWMFT4998E: An internal error occurred. The exception is java.lang.IllegalStateException: WEB#timeoutApp#timeoutApp.war
```

**Comment by Azquelt** [comment]
This was fixed back in #7247. We now have a message saying that the app has stopped if we're unable to apply the thread context, rather than that an internal error occurred and no FFDC is generated.
